xdsl/dialects/linalg.py

```diff
@@ -475,26 +465,42 @@ def parse(cls, parser: Parser):
         else:
             outs = ()
 
-        if parser.parse_optional_keyword("attrs"):
-            parser.parse_punctuation("=")
-            attrs = parser.expect(
-                parser.parse_optional_attr_dict, "expect extra attributes"
-            )
-        else:
-            attrs = {}
+        if not cls.PRINT_ATTRS_IN_FRONT:
+            if parser.parse_optional_keyword("attrs"):
+                parser.parse_punctuation("=")
+                attrs = parser.expect(
+                    parser.parse_optional_attr_dict, "expect extra attributes"
+                )
+            else:
+                attrs = {}
 
         if parser.parse_optional_punctuation("->"):
             res_types = parser.parse_optional_comma_separated_list(
                 parser.Delimiter.PAREN, parser.parse_attribute
             )
             if res_types is None:
-                res_types = [[parser.parse_attribute()]]
+                res_types = [parser.parse_attribute()]
```

Comment:
Somehow I thought the double list is needed by virtue of results being a variadic.

Reply:
Aha yes, they were put in triple brackets because of an error somewhere else, but i fixed this and put the double brackets back!